jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/tools/Jmap.java

```diff
@@ -0,0 +1,157 @@
+			+ "    -histo:live : Print only live objects%n" + "    -J: supply arguments to the Java VM running jps%n"
+			+ " NOTE: this utility may significantly affect the performance of the target JVM.%n"
+			+ "At least one option must be selected.";
+
+	/**
+	 * Print a list of Java processes and information about them.
+	 * 
+	 * @param args Arguments to the application
+	 */
+	public static void main(String[] args) {
+
+		if (!parseArguments(args)) {
+			System.exit(1);
+		}
+		AttacherDiagnosticsProvider diagProvider = new AttacherDiagnosticsProvider();
+
+		String myId = AttachHandler.getVmId();
+		for (String vmid : vmids) {
+			if (vmid.equals(myId)) {
+				continue;
+			}
+			/* if the ID looks like a process ID, check if it is running */
+			if (vmid.matches("\\d+")) { //$NON-NLS-1$
+				long pid = Long.parseLong(vmid);
+				if (!IPC.processExists(pid)) {
+					System.out.print("No such process: " + vmid); //$NON-NLS-1$
+					continue;
+				}
+			}
+			try {
+				diagProvider.attach(vmid);
+				if (histo) {
+					runAndPrintCommand(diagProvider, DiagnosticUtils.makeHeapHistoCommand(live));
+				}
+			} catch (Exception e) {
+				System.err.printf("Error getting data from %s", vmid); //$NON-NLS-1$
+				final String msg = e.getMessage();
+				if (null == msg) {
+					System.err.println();
+				} else {
+					if (msg.matches(IPC.INCOMPATIBLE_JAVA_VERSION)) {
+						System.err.println(": incompatible target JVM"); //$NON-NLS-1$
+					} else {
+						System.err.printf(": %s%n", msg); //$NON-NLS-1$
+					}
+				}
+				if (DiagnosticProperties.isDebug) {
+					e.printStackTrace();
+				}
+			} finally {
+				try {
+					diagProvider.detach();
+				} catch (IOException e) {
+					/* ignore */
+				}
+			}
+		}
+	}
+
+	private static void runAndPrintCommand(AttacherDiagnosticsProvider diagProvider, String cmd) throws IOException {
+		Properties props = diagProvider.executeDiagnosticCommand(cmd);
+		DiagnosticProperties.dumpPropertiesIfDebug("jmap result:", props); //$NON-NLS-1$
+		String responseString = new DiagnosticProperties(props).printStringResult();
+		System.out.print(responseString);
+	}
+
+	@SuppressWarnings("nls")
+	private static boolean parseArguments(String[] args) {
+		boolean okay = true;
+		vmids = new ArrayList<>();
+		histo = false;
+		live = true;
+		int optionsSelected = 0;
+		String errorMessage = "";
+		vmids = new ArrayList<>();
+		for (String a : args) {
+			if (!a.startsWith("-")) {
+				vmids.add(a);
+			} else if (a.startsWith("-histo")) {
+				histo = true;
+				optionsSelected += 1;
+				if (a.endsWith(":live")) {
+					live = true;
+				}
+			} else {
+				okay = false;
+				errorMessage = "unrecognized option " + a;
+				break;
+			}
+		}
+		if (1 != optionsSelected) {
+			okay = false;
+			errorMessage = "exactly one option must be selected";
+		}
+		if (!okay) {
+			System.err.printf("%s.%n", errorMessage); //$NON-NLS-1$
+			System.err.printf(HELPTEXT);
+			okay = false;
```

Comment:
This is redundant.

Reply:
True.  Fixed and cleaned up other redundancies.